src/drivers/intel/cavs/hda-dma.c

```diff
@@ -1005,6 +1005,81 @@ static int hda_dma_remove(struct dma *dma)
 	return 0;
 }
 
+static int hda_dma_in_data_size(struct hda_chan_data *chan)
+{
+	int size = 0;
+	uint32_t read_ptr;
+	uint32_t write_ptr;
```

Comment:
you could declare read_ptr and write_ptr inside the "if" clause. Alternatively you could invert the if like
```
if (!...)
	return 0;
...
return size;
```
which would unindent the whole function.